src/etag.rs

```diff
@@ -0,0 +1,143 @@
+//! Types for handling etags.
+use reqwest::header::HeaderValue;
+use snafu::ResultExt;
+use std::convert::TryFrom;
+
+/// Represents resources identified by etags.
+#[derive(Debug, PartialEq)]
+pub struct Etagged<T> {
+    /// A possible etag.
+    ///
+    /// It is possible, although unlikely, that a response which should contain an etag header does
+    /// not, or that etag header is invalid. In such cases this field will be `None`.
+    pub etag: Option<Etag>,
+    /// The value identified by this etag.
+    ///
+    /// This can be `None` if we have already received the data which this etag identifies.
+    pub value: Option<T>,
+}
+
+/// An etag.
+#[derive(Debug, PartialEq)]
+pub struct Etag(String);
```

Comment:
I think it would be more useful to re-export the `hyperx::header::ETag` than creating our own type as we already depend on it for handling headers in pages. https://docs.rs/hyperx/1.2.1/hyperx/header/struct.ETag.html

Reply:
Completely overlooked this, this is much nicer!